### PR TITLE
Implement GetDepositContract in the config API

### PR DESCRIPTION
**Pull request by rkapka**
**What type of PR is this?**

Feature

What does this PR do? Why is it needed?

This PR implements config API's `GetDepositContract` method according to the spec: https://ethereum.github.io/eth2.0-APIs/#/Config/getDepositContract

Which issues(s) does this PR fix?

Part of #7510

Other notes for review

N/A

### CODE REVIEW

File: beacon-chain/rpc/beaconv1/config.go [hunk 1]
```diff
@@ -23,5 +30,21 @@ func (bs *Server) GetSpec(ctx context.Context, req *ptypes.Empty) (*ethpb.SpecRe
...
 func (bs *Server) GetDepositContract(ctx context.Context, req *ptypes.Empty) (*ethpb.DepositContractResponse, error) {
-	return nil, errors.New("unimplemented")
+	ctx, span := trace.StartSpan(ctx, "beaconv1.GetDepositContract")
+	defer span.End()
+
+	buf := bytes.NewBuffer(params.BeaconConfig().GenesisForkVersion)
+	chainId, err := binary.ReadUvarint(buf)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "could not obtain genesis fork version: %v", err)
+	}
```
Comment (rkapka) [inline beacon-chain/rpc/beaconv1/config.go:40]: I did not test the error case as this can only happen when our fork version is completely broken.

File: beacon-chain/rpc/beaconv1/config.go [hunk 2]
```diff
@@ -23,5 +30,21 @@ func (bs *Server) GetSpec(ctx context.Context, req *ptypes.Empty) (*ethpb.SpecRe
...
+	byteAddress := [20]byte(bs.PowchainInfoFetcher.DepositContractAddress())
```
Comment (terencechain) [inline beacon-chain/rpc/beaconv1/config.go:41]: Similar to my last feedback, we do not need to implement a new service getter to obtain deposit contract address. This is already accessible via `params.BeaconConfig().DepositContractAddress` (where powchain service gets it too)

**Review by terencechain — CHANGES_REQUESTED** [review]

File: beacon-chain/rpc/beaconv1/config.go (hunk 2 above)
Comment (rkapka) [reply beacon-chain/rpc/beaconv1/config.go:41]: Good catch. I have to go through the config file and make a mental note of what is available.

**Comment by rkapka** [comment]
Actually it turned out I tried to fetch the wrong thing. It was supposed to be the Eth1 chain ID, not the Eth2 fork version. There was a mistake in the endpoint's description. I fixed it in one of the commits.

File: beacon-chain/rpc/beaconv1/server.go [hunk 3]
```diff
@@ -32,6 +32,7 @@ type Server struct {
 	ChainInfoFetcher    blockchain.ChainInfoFetcher
 	DepositFetcher      depositcache.DepositFetcher
 	BlockFetcher        powchain.POWBlockFetcher
+	PowchainInfoFetcher powchain.ChainInfoFetcher
```
Comment (terencechain) [inline beacon-chain/rpc/beaconv1/server.go:35]: Are these changes still relevant to this PR?

Comment (rkapka) [reply beacon-chain/rpc/beaconv1/server.go:35]: Nope. +1

**Review by terencechain — APPROVED** [review]
